v2/spanner-to-sourcedb/README.md

```diff
@@ -0,0 +1,363 @@
+
+Migrating a database is a complex affair, involving changes to the schema, converting the application, tuning for performance, ensuring minimal downtime and completeness during data migration. It is possible that after migration and cutover, issues/inconsistent performance are encountered on the target (Cloud Spanner) requiring a fallback to the original source database with minimal disruption to the service. Reverse replication enables this fallback by replicating data written on Cloud Spanner back to the source database.This allows the application to point to the source and continue serving requests correctly.
+
+Reverse replication could also be used to replicate the Cloud Spanner writes to a different database, other than the source database, for performing reconciliation, validations and reporting.
+
+### How it works
+
+Reverse replication flow involves below steps:
+
+1. Reading the changes that happened on Cloud Spanner using [Cloud Spanner change streams](https://cloud.google.com/spanner/docs/change-streams)
+2. Removing forward migrated changes ( if configured to filter )
+3. Transforming Cloud Spanner data to source database schema
+4. Verifying that the source database does not already have more up-to-date data for the given primary key
+5. Writing to source database
+
+These steps are achieved by the Spanner to SourceDb dataflow template.
+
+![Architecture](https://services.google.com/fh/files/misc/revreplpko.png)
+
+#### Consistency guarantees
+
+The pipeline guarantees consistency at a primary key level. The pipeline creates shadow tables, which contain the Spanner commit timestamp of the latest record that was successfully written on the shard for that table. The writes are guaranteed to be consistent up-to this commit timestamp for the given primary key.
+
+However, there is no order maintained across tables or even across various primary keys in the same table. This helps achieve faster replication.
+
+The reverse replication is eventually consistent. Meaning once there are no retryable or severe errors and all records from Spanner have been successfully replicated, the source database reaches consistent state.
+
+#### Run modes
+
+The Dataflow job that handles reverse replication runs in two modes:
+
+- **regular**: This is the default mode, where the events streamed by Spanner change streams are picked up and converted to source compatible data types and applied to source database. It also does automatic retry of retryable errors and once the retry is exhausted, moves them to the 'severe' folder of the dead letter queue (DLQ) directory in GCS. Permanent errors are also moved to the 'severe' folder of dead letter queue.
+
+- **retryDLQ**: This mode looks at the 'severe' folder of DLQ and retries the events. This mode is ideal to run when all the permanent and/or retryable errors are fixed - for example any bug fix/ dependent data migration is complete.This mode only reads from DLQ and not from Spanner change streams.
```

Comment:
`This mode only reads from DLQ and not from Spanner change streams.` might give an idea that retry folder events are also retried.

Reply:
Severe records are picked up - if they go into retry - they are also retired.

Reply:
The wording isn't clear enough, i think it will be better to add 1 line stating this.

Reply:
Thanks Shreya! Done